### PR TITLE
#378: changed /check/alive endpoint from HTTP HEAD only to HTTP GET

### PR DESCRIPTION
**Pull request by 4w31ss3**
Changes check/alive from HTTP HEAD to GET method for Kubernetes LivenessProbe

--- 
<sup>Albert Weißert <albert.weissert@daimler.com>, Daimler TSS GmbH, [imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sup>

### CODE REVIEW

File: sechub-server-core/src/main/java/com/daimler/sechub/server/core/AnonymousCheckAliveRestController.java [hunk 1]
```diff
@@ -19,9 +20,10 @@
 				name="REST API call",
 				needsRestDoc=true,
 				description="An anonymous user checks if the server is alive and running using the REST API"))
-	@RequestMapping(path = APIConstants.API_ANONYMOUS + "check/alive", method = RequestMethod.HEAD)
-	public void checkAlive() {
-	    /* do nothing here - its just a HEAD request */
+	@RequestMapping(path = APIConstants.API_ANONYMOUS + "check/alive", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE})
```
Comment (Jeeppler) [inline sechub-server-core/src/main/java/com/daimler/sechub/server/core/AnonymousCheckAliveRestController.java:23]: Please map the `RequestMethod` explicitly to `GET` and `HEAD`: https://stackoverflow.com/a/3803882

`GET` is mapped to `HEAD` implicitly by Spring MVC: https://docs.spring.io/spring-framework/docs/current/spring-framework-reference/web.html#mvc-ann-requestmapping-head-options

Comment (de-jcup) [reply sechub-server-core/src/main/java/com/daimler/sechub/server/core/AnonymousCheckAliveRestController.java:23]: @4w31ss3 please do as @Jeeppler  suggested: please add both to RequestMethod so it is 
1. it is more clear that we support both
2. api changes will not affect us

Comment (4w31ss3) [reply sechub-server-core/src/main/java/com/daimler/sechub/server/core/AnonymousCheckAliveRestController.java]: @de-jcup @Jeeppler done

**Comment by de-jcup** [comment]
Review has been done, PR will be merged

---
<sup>Albert Tregnaghi <albert.tregnaghi@daimler.com>, Daimler TSS GmbH, [imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sup>

**Pull request merged**